orderer/consensus/smartbft/chain.go

```diff
@@ -0,0 +1,604 @@
+/*
```

Comment:
this code has no test. Let's open an issue to write a unit test for this file.

Reply:
https://github.com/hyperledger/fabric/issues/4008